compiler/src/dotty/tools/dotc/semanticdb/TypeOps.scala

```diff
@@ -0,0 +1,304 @@
+package dotty.tools
+package dotc
+package semanticdb
+
+import core.Symbols._
+import core.Contexts.Context
+import core.Types._
+import core.Annotations.Annotation
+import core.Flags
+import core.Names.Name
+import core.StdNames.tpnme
+import ast.tpd._
+
+import collection.mutable
+
+import dotty.tools.dotc.{semanticdb => s}
+
+class TypeOps:
+  import SymbolScopeOps._
+  private val paramRefSymtab = mutable.Map[(LambdaType, Name), Symbol]()
+  private val refinementSymtab = mutable.Map[(RefinedType, Name), Symbol]()
+  given typeOps: TypeOps = this
```

Comment:
Because we pass the instance of `TypeOps` to `sscope` (which pass it to `symbolInfo`)
Maybe we should give the instance of `TypeOps` explicitly because I use `given` for no big reason: I used `given` just because I was too lazy 😅